<launch>
    <arg name="namespace"                          default="drone1" />
    <arg name="frecuency"                          default="100.0" />
    <arg name="activate_behavior_srv"              default="activate_behavior" />
    <arg name="deactivate_behavior_srv"            default="deactivate_behavior" />
    <arg name="check_activation_conditions_srv"    default="check_activation_conditions" />
    <arg name="activation_finished_topic"          default="behavior_activation_finished" />
    <arg name="behavior_system"                    default="swarm_interaction" />
    <arg name="social_comunication_channel_topic"  default="social_comunication_channel" />
    <arg name="add_belief_srv"                     default="add_belief" />
    <arg name="message_from_robot"		   default="message_from_robot" />
    <arg name="shared_robot_positions_channel"     default="shared_robot_positions_channel" />
    <arg name="pose_topic"		           default="self_localization/pose" />
 

   <group ns="$(arg namespace)/$(arg behavior_system)">
	     <param name="~namespace"                           value="$(arg namespace)"                          type="str" />
           <param name="~frecuency"                           value="$(arg frecuency)"                          type="double" />
             <param name="~activate_behavior_srv"               value="$(arg activate_behavior_srv)"              type="str" />
             <param name="~deactivate_behavior_srv"             value="$(arg deactivate_behavior_srv)"            type="str" />
             <param name="~check_activation_conditions_srv"     value="$(arg check_activation_conditions_srv)"    type="str" />
             <param name="~shared_robot_positions_channel"      value="$(arg shared_robot_positions_channel)"     type="str" />
             <param name="~social_comunication_channel_topic"    value="$(arg social_comunication_channel_topic)"  type="str" />
             <param name="~add_belief_srv"                      value="$(arg add_belief_srv)"                     type="str" />
             <param name="~behavior_system"                     value="$(arg behavior_system)"                    type="str" />
             <param name="~message_from_robot"                   value="$(arg message_from_robot)"                 type="str" />
             <param name="~pose_topic"                          value="$(arg pose_topic)"                         type="str" />
           
       <node pkg="nodelet" type="nodelet" args="manager" name="swarm_interaction_manager" output="screen" />
       <node pkg="nodelet" type="nodelet" args="load swarm_interaction/BehaviorInformRobots swarm_interaction_manager"
        name="behavior_inform_robots" output="screen"/>
       <node pkg="nodelet" type="nodelet" args="load swarm_interaction/BehaviorPayAttentionToRobotMessages swarm_interaction_manager"
        name="behavior_pay_attention_to_robot_messages" output="screen"/>
       <node pkg="nodelet" type="nodelet" args="load swarm_interaction/BehaviorInformPositionToRobots swarm_interaction_manager"
        name="behavior_inform_position_to_robots" output="screen"/>
 
  </group>
</launch>
